R/excitation.R

```diff
@@ -0,0 +1,147 @@
+#' Excitation
+#'
+#' @description Excitation lets you add visual citations as tooltips to your
+#'   \pkg{xaringan} slides. Simply include a `.bib` file in your R Markdown
+#'   YAML header, as usual, and cite away!
+#'
+#' @section Usage: To use excitation for visual citations, add the following
+#'   code chunk to your slides' R Markdown file.
+#'
+#'   ````markdown
+#'   ```{r xaringan-excitation, echo=FALSE}
+#'   xaringanExtra::use_excitation()
+#'   ```
+#'   ````
+#'
+#' @examples
+#' use_excitation()
+#'
+#' @name excitation
+NULL
+
+#' @describeIn excitation Adds visual citations to your xaringan slides.
+#' @export
+use_excitation <- function() {
+  assign('.excitation_keys', vector('character'), envir = globalenv())
```

Comment:
Instead of using the global env, you need to create an internal environment in the package namespace where we can stash these keys, e.g.

```r
excitation_keys <- new.env(parent = emptyenv())

use_execitation <- function() {
  # put keys in `excitation_keys` as needed
}
```

You'll also need to check that this still works as expected when rendering interactively with `rmarkdown::render()`.